### DISCUSSION TITLE
PostProcess tensile_test Example to Generate Stress Vs Strain Plot

### DISCUSSION THREAD

**Discussion by ppandit95** [peridigm/peridigm#360] [Q&A] [answered]
Dear Dr @djlittl 

Although this querry of mine seems quite trivial but I would like to know the procedure to run the blot script followed by generation of desired plot using GNUPLOT.

Thanks in Advance
Pushkar

**Comment by djlittl** [comment, marked as answer]
The commands are in the file tensile_test.post

The commands are:
blot tensile_test.e -input tensile_test.blot cps
gnuplot tensile_test.plotgen
tensile_test_compute_modulus.py

For this to work, you'll need blot and gnuplot installed.

The general idea here is to extract a few global variables from the tensile_test.e file that Peridigm writes, and then construct a stress-strain plot.  The global variables are the reaction forces on the top and bottom of the tensile bar, and the locations of a "virtual strain gauge", which just tracks the deformed and undeformed locations of a pair of points on the tensile bar (which is what a real strain gauge would do).  There's no real need to use blot and gnuplot, you could do this with other post-processing tools too.

The python script in the final command just computes the modulus, which in theory should match what we gave to Peridigm as input.

**Reply by ppandit95** [reply]
Thanks alot @djlittl for your quick response to this query
